estate/models/estate_property.py

```diff
@@ -0,0 +1,99 @@
+from dateutil.relativedelta import relativedelta
+from odoo import api, models, fields
+from odoo.exceptions import UserError
+from odoo.tools.float_utils import float_compare, float_is_zero
+
+
+class EstateProperty(models.Model):
+    _name = "estate.property"
+    _description = "estate property model"
+    _order = "id desc"
+
+    name = fields.Char(required=True)
+    description = fields.Text()
+    postcode = fields.Char()
+    date_availability = fields.Date(copy=False, default=fields.Date.today() + relativedelta(months=+3))
+    expected_price = fields.Float(required=True, default=1)
+    selling_price = fields.Float(readonly=True, copy=False)
+    bedrooms = fields.Integer(default=2)
+    living_area = fields.Integer()
+    facades = fields.Integer()
+    garage = fields.Boolean()
+    garden = fields.Boolean()
+    garden_area = fields.Integer()
+    garden_orientation = fields.Selection(
+        selection=[('north', 'North'), ('south', 'South'), ('east', 'East'), ('west', 'West')]
+    )
+    active = fields.Boolean(default=True)
+    state = fields.Selection(
+        selection=[
+            ('new', 'New'),
+            ('offer_received', 'Offer Received'),
+            ('offer_accepted', 'Offer Accepted'),
+            ('sold', 'Sold'),
+            ('cancelled', 'Cancelled')
+        ],
+        required=True,
+        copy=False,
+        default='new'
+    )
+    property_type_id = fields.Many2one("estate.property.type", string="Property Type")
+    buyer_id = fields.Many2one("res.partner", copy=False)
+    salesperson_id = fields.Many2one("res.users", default=lambda self: self.env.user)
+    property_tag_ids = fields.Many2many("estate.property.tag", string="Propert Tags")
+    offer_ids = fields.One2many('estate.property.offer', 'property_id', string="Offers")
+
+    total_area = fields.Float(compute="_compute_total_area")
+    best_price = fields.Float(compute="_compute_best_price")
+
+    _positive_expected_price = models.Constraint(
```

Comment:
Should start with check as it is a constraint. It is an Odoo convention.
```suggestion
    _check_positive_expected_price = models.Constraint(
```